examples/flutter_gallery/lib/demo/pan_and_zoom_demo_transform_interaction.dart

```diff
@@ -0,0 +1,469 @@
+import 'package:flutter/material.dart';
+import 'package:vector_math/vector_math_64.dart' show Vector3;
+import 'pan_and_zoom_demo_inertial_motion.dart';
+
+// This widget allows 2D transform interaction on its child. The user can drag
+// to pan, pinch to zoom and rotate, and still get back untransformed
+// coordinates for targeted tap events.
+@immutable
+// TODO(justinmc): Need a better name. The plan is to make this a full blown
+// reusable Flutter widget.
+class TransformInteraction extends StatefulWidget {
+  const TransformInteraction({
+    // The child to perform the transformations on
+    @required this.child,
+    // The desired size of the area that can receive events
+    @required this.size,
+    // The scale will be clamped to between these values
+    this.maxScale = 2.5,
+    this.minScale = 0.8,
+    // Transforms will be limited so that the viewport can not view beyond this
+    // Rect. The Rect does not rotate with the rest of the scene, so it is
+    // always aligned with the viewport.
+    this.visibleRect,
```

Comment:
Maybe visible isn't the right word. If you zoom in and pan to a corner, most of the area inside this 'rect' is not visible. Maybe pannableViewportRect?

Reply:
Good point, that's confusing.  `boundaryRect` maybe?  Because it should constrain rotate and scale transformations as well, not just panning.

Reply:
boundary seems ok too